flang/lib/Lower/IO.cpp

```diff
@@ -921,6 +921,105 @@ constexpr bool isDataTransferNamelist<Fortran::parser::PrintStmt>(
   return false;
 }
 
+/// Lowers a format statment that uses a assigned varible label reference as
+/// a select operation to allow for run-time selection of the format statement
+// Possible TODO: Instead of inlining a selectOp every time there is a format
+// statement a function with the selectOpcould be generated to reduce code size
+static std::tuple<mlir::Value, mlir::Value, mlir::Value>
+lowerReferenceAsStringSelect(Fortran::lower::AbstractConverter &converter,
+                             mlir::Location loc,
+                             const Fortran::evaluate::Expr
+                             <Fortran::evaluate::SomeType> &expr,
+                             Fortran::lower::pft::LabelEvalMap *labelMap,
+                             Fortran::lower::pft::SymbolLabelMap &assignMap,
+                             Fortran::lower::SymMap &localSymbols,
+                             mlir::Type strTy, mlir::Type lenTy) {
+
+  // Create the requisite blocks to inline a selectOp
+  auto &builder = converter.getFirOpBuilder();
+  auto* startBlock = builder.getBlock();
+  auto* endBlock = startBlock->splitBlock(builder.getInsertionPoint());
+  auto* block = startBlock->splitBlock(builder.getInsertionPoint());
+  builder.setInsertionPointToEnd(block);
+
+  llvm::SmallVector<int64_t, 5> indexList;
+  llvm::SmallVector<mlir::Block *, 5> blockList;
+  auto stringRefType = builder.getRefType(
+    fir::SequenceType::get({fir::SequenceType::getUnknownExtent()},
+                           fir::CharacterType::get(builder.getContext(), 1)));
+  auto symbol = GetLastSymbol(&expr);
+  auto labelIter = assignMap.find(*symbol);
+  assert(labelIter != assignMap.end() && "ICE: Label not found in map");
+  auto labels = labelIter->second;
+
+  for (auto label : labels) {
+    indexList.push_back(label);
+    auto iter = labelMap->find(label);
+    assert(iter != labelMap->end() && "ICE: Label is missing from the table");
+
+    // Extract the inner text from the format statement
```

Comment:
Why not using `lowerSourceTextAsStringLit` here ?

Reply:
Originally I had a reason but it no longer applies, I will change that.